### PR TITLE
chore(snapshots): persist chekcksum with snapshots

### LINKED ISSUE #6290: Add checksum file to the snapshot to detect partial snapshots
**Description**

Follow up #6255

Snapshot directory can have only a partial set of files due to various reasons such as - incorrect handling of concurrent access, deleting snapshot failed, system crashed while moving the snapshot when atomic move not available etc.

Snapshot should contain a checksum file. The checksum should be calculated immediately after the snapshot was taken. If it was partially moved or partially deleted, we will be able to detect it by verifying the checksum.

### PR DESCRIPTION
**Pull request by deepthidevaki**
## Description

* Calculate a checksum when taking the snapshot
* Persist the checksum in the snapshot directory
* On startup verify the checksum before loading the snapshots
* If there are no snapshots available after a restart due to corruption and the log has been compacted, then it fails the raft startup.

## Related issues

closes #6290 

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/zeebe-io/zeebe/compare/stable/0.24...develop?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/0.25`) to the PR, in case that fails you need to create backports manually.

Testing:
* [x] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark 

Documentation: 
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] New content is added to the [release announcement](https://drive.google.com/drive/u/0/folders/1DTIeswnEEq-NggJ25rm2BsDjcCQpDape)

### CODE REVIEW

**Comment by deepthidevaki** [comment]
@MiguelPires Some tests are failing. Let me fix it before you review.

**Comment by deepthidevaki** [comment]
@MiguelPires It is ready to review.

File: snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedTransientSnapshot.java [hunk 1]
```diff
@@ -75,6 +77,15 @@ private void takeInternal(final Predicate<Path> takeSnapshot) {
     }
   }
 
+  private void calculateAndPersistChecksum() throws IOException {
+    if (!directory.toFile().exists() || directory.toFile().listFiles().length == 0) {
+      // Do not calculate checksum if directory is empty
```
Comment (MiguelPires) [inline snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedTransientSnapshot.java:82]: Shouldn't the snapshot be invalid if there are no files?

Comment (deepthidevaki) [reply snapshot/src/main/java/io/zeebe/snapshots/broker/impl/FileBasedTransientSnapshot.java:82]: I thought about it. But was not sure what should be the semantics. I will change the semantics to mark the snapshot invalid if there are no files.

**Comment by deepthidevaki** [comment]
@MiguelPires I have update the PR. The snapshot is now marked as invalid immediately after it is taken if there are no files.

**Review by miguelpires — APPROVED** [review]
Looks good @deepthidevaki, thanks

**Pull request merged**